src/webportal/src/app/home/home/abnormal-job-list.jsx

```diff
@@ -0,0 +1,237 @@
+        if (isLowGpuUsageJob(job)) {
+          return (<div style={{color: palette.red}}>{`count: ${job.totalGpuNumber}  usage: ${job.gpuUsage}%`}</div>);
+        }
+        return job.totalGpuNumber;
+      },
+    },
+    {
+      key: 'modified',
+      minWidth: 150,
+      name: 'Date Modified',
+      className: FontClassNames.mediumPlus,
+      headerClassName: FontClassNames.medium,
+      isResizable: true,
+      onRender(job) {
+        return getJobModifiedTimeString(job);
+      },
+    },
+    {
+      key: 'user',
+      minWidth: 60,
+      name: 'User',
+      fieldName: 'username',
+      className: FontClassNames.mediumPlus,
+      headerClassName: FontClassNames.medium,
+      isResizable: true,
+    },
+    {
+      key: 'duration',
+      minWidth: 120,
+      name: 'Duration',
+      className: FontClassNames.mediumPlus,
+      headerClassName: FontClassNames.medium,
+      isResizable: true,
+      onRender(job) {
+        if (isLongRunJob(job)) {
+          return (<div style={{color: palette.red}}>{getJobDurationString(job)}</div>);
+        }
+        return getJobDurationString(job);
+      },
+    },
+    {
+      key: 'virtualCluster',
+      minWidth: 100,
+      name: 'Virtual Cluster',
+      fieldName: 'virtualCluster',
+      className: FontClassNames.mediumPlus,
+      headerClassName: FontClassNames.medium,
+      isResizable: true,
+    },
+    {
+      key: 'status',
+      minWidth: 100,
+      name: 'Status',
+      headerClassName: FontClassNames.medium,
+      isResizable: true,
+      onRender(job) {
+        return <StatusBadge status={getHumanizedJobStateString(job)} />;
+      },
+    },
+    {
+      key: 'action',
+      minWidth: 100,
+      name: 'Action',
+      headerClassName: FontClassNames.medium,
+      className: zeroPaddingClass,
+      isResizable: true,
+      onRender(job) {
+        return (
+          <div
+            style={{
+              height: '100%',
+              display: 'flex',
+              alignItems: 'center',
+              justifyContent: 'center',
+            }}
+            data-selection-disabled
+          >
+            <DefaultButton
+              iconProps={{iconName: 'StopSolid'}}
+              styles={{
+                root: {backgroundColor: '#e5e5e5'},
+                rootFocused: {backgroundColor: '#e5e5e5'},
+                rootDisabled: {backgroundColor: '#eeeeee'},
+                rootCheckedDisabled: {backgroundColor: '#eeeeee'},
+                icon: {fontSize: 12},
+              }}
+              onClick={(e) => {
+                e.stopPropagation();
+                stopJob(job);
+              }}
+            >
+              Stop
+            </DefaultButton>
+          </div>
+        );
+      },
+    },
+  ];
+
+  const stopJob = useCallback((job) => {
+    userAuth.checkToken((token) => {
+      const {name, username} = job;
+      fetch(`${webportalConfig.restServerUri}/api/v1/jobs/${username}~${name}/executionType`, {
```

Comment:
I think we should extract the api call function to conn.js

Reply:
I will extract it to conn.js. Actually I think we need to extract these common APIs to a common lib. Maybe JS SDK is doing such things.